src/steamship/agents/functional/functions_based.py

```diff
@@ -0,0 +1,60 @@
+from typing import List
+
+from steamship import Block
+from steamship.agents.openai.output_parser import FunctionsBasedOutputParser
+from steamship.agents.schema import (
+    Action,
+    AgentContext,
+    ConversationalLLM,
+    ConversationalLLMAgent,
+    Tool,
+)
+from steamship.data.tags.tag_constants import RoleTag
+
+
+class FunctionsBasedAgent(ConversationalLLMAgent):
+    """Selects actions for AgentService based on OpenAI Function style LLM Prompting."""
+
+    PROMPT = """You are a helpful AI assistant. Only use the functions you have been provided with.
+
+NOTE: Some functions return images, video, and audio files. These multimedia files will be represented in messages as
+UUIDs for Steamship Blocks. When responding directly to a user, you SHOULD print the Steamship Blocks for the images,
+video, or audio as follows: `Block(UUID for the block)`.
+
+Example response for a request that generated an image:
+Here is the image you requested: Block(288A2CA1-4753-4298-9716-53C1E42B726B)."""
```

Comment:
Wow this is so much shorter

Reply:
🔥

Reply:
I'm super curious to hear how variance-tolerant you've found the prompt to be -- I think we've all wrestled with the ReAct one in the past.